### PR TITLE
Make assumptions on mass Operator in InstationaryModel consistent

### LINKED ISSUE #782: InstationaryModel mass operator (non)parametric
[Documentation](https://github.com/pymor/pymor/blob/5705ddcc6df13aa4f5424382e55cf2f858e73a5d/src/pymor/models/basic.py#L160-L161) says it has to be time-independent and parameter-independent. In [`__init__`](https://github.com/pymor/pymor/blob/5705ddcc6df13aa4f5424382e55cf2f858e73a5d/src/pymor/models/basic.py#L235), it is passed to `build_parameter_type`. [Also](https://github.com/pymor/pymor/blob/5705ddcc6df13aa4f5424382e55cf2f858e73a5d/src/pymor/models/basic.py#L228-L229), it is not checked if `mass` is parametric.

### PR DESCRIPTION
**Pull request by sdrave**
We allow `mass` to be `parametric` and depend on time.

Fixes #782.

### CODE REVIEW

File: src/pymor/algorithms/timestepping.py [hunk 1]
```diff
@@ -150,16 +150,14 @@ def implicit_euler(A, F, M, U0, t0, t1, nt, mu=None, num_values=None, solver_opt
...
-
     R = A.source.empty(reserve=nt+1)
     R.append(U0)
 
     options = A.solver_options if solver_options == 'operator' else \
               M.solver_options if solver_options == 'mass' else \
               solver_options
     M_dt_A = (M + A * dt).with_(solver_options=options)
-    if not A_time_dep:
+    if not M_dt_A.parametric or '_t' not in M_dt_A.solver_options:
```
Comment (pmli) [inline src/pymor/algorithms/timestepping.py]: Maybe this?
```suggestion
    if not M_dt_A.parametric or '_t' not in M_dt_A.parameter_type:
```

File: src/pymor/reductors/parabolic.py [hunk 2]
```diff
@@ -53,7 +53,10 @@ class ParabolicRBReductor(InstationaryRBReductor):
     """
     def __init__(self, fom, RB=None, product=None, coercivity_estimator=None,
                  check_orthonormality=None, check_tol=None):
-        assert isinstance(fom.time_stepper, ImplicitEulerTimeStepper)
+        if not isinstance(fom.time_stepper, ImplicitEulerTimeStepper):
+            raise NotImplementedError
+        if fom.mass is not None and 't_' in fom.mass.parameter_type:
```
Comment (pmli) [inline src/pymor/reductors/parabolic.py]: ?
```suggestion
        if fom.mass is not None and '_t' in fom.mass.parameter_type:
```

File: src/pymor/algorithms/timestepping.py (hunk 1 above)
Comment (sdrave) [reply src/pymor/algorithms/timestepping.py]: Whoops.

File: src/pymor/reductors/parabolic.py (hunk 2 above)
Comment (sdrave) [reply src/pymor/reductors/parabolic.py]: Well, this was a high-quality PR ..

**Pull request merged**